### PR TITLE
implement v4

### PR DESCRIPTION
**Pull request by topi314**
- [x] Removal of deprecated endpoints in v3.7.0 (all non /v3 prefixed endpoints except /version)
- [x] Replacing the resumeKey with the sessionId to resume a Lavalink session
- [x] Removal of all json fields which were deprecated in v3 (track in events, etc.)
- [x] Addition of artworkUrl and isrc in the track json & encoded tracks
- [x] Addition to the plugin api to be able modify the track json. See https://github.com/freyacodes/Lavalink/pull/788 for more info
- [x] Addition of the full track json in all TrackXEvents

### CODE REVIEW

**Review by freyacodes — APPROVED** [review]

File: plugin-api/src/main/java/dev/arbjerg/lavalink/api/AudioPluginInfoModifier.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import com.sedmelluq.discord.lavaplayer.track.AudioPlaylist;
+import com.sedmelluq.discord.lavaplayer.track.AudioTrack;
+
+public interface AudioPluginInfoModifier {
+
+	/**
+	 * Adds custom fields to an {@link AudioTrack}'s JSON.
+	 *
+	 * @param track the track that was loaded.
+	 * @param node the info node which can be altered.
```
Comment (freyacodes) [inline plugin-api/src/main/java/dev/arbjerg/lavalink/api/AudioPluginInfoModifier.java]:
```suggestion
	 * @param node the JSON object which can be altered.
```

File: plugin-api/src/main/java/dev/arbjerg/lavalink/api/AudioPluginInfoModifier.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	 */
+	default void modifyAudioTrackPluginInfo(AudioTrack track, ObjectNode node) {
+	}
+
+	/**
+	 * Adds custom fields to an {@link AudioPlaylist}'s JSON.
+	 *
+	 * @param playlist the playlist that was loaded.
+	 * @param node the info node which can be altered.
```
Comment (freyacodes) [inline plugin-api/src/main/java/dev/arbjerg/lavalink/api/AudioPluginInfoModifier.java]:
```suggestion
	 * @param node the JSON object which can be altered.
```

**Review by freyacodes — APPROVED** [review]
I only have these docs to nitpick on

**Comment by ooliver1** [comment]
`/metrics` seems to suddenly require `Authorization`
```
2023-05-04 10:31:54.107  WARN 1 --- [  XNIO-1 task-1] l.s.config.RequestAuthorizationFilter    : Authorization failed for 127.0.0.1 on GET /metrics
```

**Comment by topi314** [comment]
Known issues:
* `/v4/decodetracks` currently doesn't work

**Comment by topi314** [comment]
lavalink v4 now requires java 17

**Pull request merged**